core/testcontainers/core/docker_client.py

```diff
@@ -154,3 +160,28 @@ def _stop_container(container: Container) -> None:
         pass
     except Exception as ex:
         LOGGER.warning("failed to shut down container %s with image %s: %s", container.id, container.image, ex)
+
+
+def _register_signal_handler(handler: Callable[[], None], signals: Iterable) -> None:
```

Comment:
A more advanced version of `signal.signal` that does not overwrite existing signal handler registered to signal.